cl/search/documents.py

```diff
@@ -1070,3 +1081,578 @@ def prepare(self, instance):
+
+    def prepare_caseNameFull(self, instance):
+        return instance.cluster.case_name_full
+
+    def prepare_dateFiled_text(self, instance):
+        if not instance.cluster.date_filed:
+            return
+
+        if isinstance(instance.cluster.date_filed, str):
+            return datetime.strptime(
+                instance.cluster.date_filed, "%Y-%m-%d"
+            ).strftime("%-d %B %Y")
+
+        return instance.cluster.date_filed.strftime("%-d %B %Y")
+
+    def prepare_dateArgued_text(self, instance):
+        if instance.cluster.docket.date_argued:
+            return instance.cluster.docket.date_argued.strftime("%-d %B %Y")
+
+    def prepare_dateReargued_text(self, instance):
+        if instance.cluster.docket.date_reargued:
+            return instance.cluster.docket.date_reargued.strftime("%-d %B %Y")
+
+    def prepare_dateReargumentDenied_text(self, instance):
+        if instance.cluster.docket.date_reargument_denied:
+            return instance.cluster.docket.date_reargument_denied.strftime(
+                "%-d %B %Y"
+            )
+
+    def prepare_court_id(self, instance):
+        return instance.cluster.docket.court.pk
+
+    def prepare_court(self, instance):
+        return instance.cluster.docket.court.full_name
+
+    def prepare_court_citation_string(self, instance):
+        return instance.cluster.docket.court.citation_string
+
+    def prepare_judge(self, instance):
+        return instance.cluster.judges
+
+    def prepare_panel_names(self, instance):
+        return [judge.name_full for judge in instance.cluster.panel.all()]
+
+    def prepare_citation(self, instance):
+        return [str(cite) for cite in instance.cluster.citations.all()]
+
+    def prepare_attorney(self, instance):
+        return instance.cluster.attorneys
+
+    def prepare_suitNature(self, instance):
+        return instance.cluster.nature_of_suit
+
+    def prepare_status(self, instance):
+        return instance.cluster.get_precedential_status_display()
+
+    def prepare_proceduralHistory(self, instance):
+        return instance.cluster.procedural_history
+
+    def prepare_posture(self, instance):
+        return instance.cluster.posture
+
+    def prepare_syllabus(self, instance):
+        return instance.cluster.syllabus
+
+    def prepare_cluster_child(self, instance):
+        parent_id = getattr(instance.cluster, "pk", None)
+        return {"name": "opinion", "parent": parent_id}
+
+
+@opinion_index.document
+class OpinionClusterDocument(OpinionBaseDocument):
+    court_exact = fields.KeywordField(attr="docket.court_id")
+    absolute_url = fields.KeywordField()
+    caseNameShort = fields.TextField(
+        attr="case_name_short",
+        analyzer="text_en_splitting_cl",
+        fields={
+            "exact": fields.TextField(analyzer="english_exact"),
+        },
+        search_analyzer="search_analyzer",
+    )
+    non_participating_judge_ids = fields.ListField(
```

Comment:
Also, will this one and `source` be available to be queried only on the `OpinionClusterDocument`, with no combination of child fields?

Reply:
Neither this field nor `source` appears in the list provided on the [Advanced Query Techniques and Operators page](https://www.courtlistener.com/help/search-operators/). I've added both of them because they are included in the SOLR mapping.

Reply:
Got it, yeah in this case I think we should confirm we don't to enable these fields to be searchable in combination of child fields.